agent/engine/docker_task_engine.go

```diff
@@ -649,6 +649,19 @@ func (engine *DockerTaskEngine) deleteTask(task *apitask.Task) {
 		}
 	}
 
+	for _, container := range task.Containers {
```

Comment:
Hmm should we clean up container resources before task resources here? or the order does not really matter here?

Reply:
Don't think it would matter since container is not removed before we clean up the container resource and they are not dependent..